pkg/types/relayer.go

```diff
@@ -100,10 +98,29 @@ type ChainService interface {
 	Replay(ctx context.Context, fromBlock string, args map[string]any) error
 }
 
+type EVMService interface {
+	CallContract(ctx context.Context, msg *evm.CallMsg, confidence primitives.ConfidenceLevel) ([]byte, error)
+	GetLogs(ctx context.Context, filterQuery evm.FilterQuery) ([]*evm.Log, error)
+	BalanceAt(ctx context.Context, account string, blockNumber *big.Int) (*big.Int, error)
+	EstimateGas(ctx context.Context, call *evm.CallMsg) (uint64, error)
+	TransactionByHash(ctx context.Context, hash string) (*evm.Transaction, error)
+	TransactionReceipt(ctx context.Context, txHash string) (*evm.Receipt, error)
+
+	// GetTransactionFee retrieves the fee of a transaction in wei from the underlying chain's TXM
+	GetTransactionFee(ctx context.Context, transactionID string) (*TransactionFee, error)
+	LatestAndFinalizedHead(ctx context.Context) (latest Head, finalized Head, err error)
+	QueryLogsFromCache(ctx context.Context, filterQuery []query.Expression,
+		limitAndSort query.LimitAndSort, confidenceLevel primitives.ConfidenceLevel) ([]*evm.Log, error)
+	RegisterLogTracking(ctx context.Context, filter evm.LPFilterQuery) error
+	UnregisterLogTracking(ctx context.Context, filterName string) error
+	GetTransactionStatus(ctx context.Context, transactionID string) (TransactionStatus, error)
```

Comment:
Do we have duplicate `TransactionStatus`s?
```suggestion
	GetTransactionStatus(ctx context.Context, transactionID string) (evm.TransactionStatus, error)
```

Reply:
I should have put a TODO for another PR that will deduplicate TransactionStatus and TransactionFee, since now they are used by TxManager

Reply:
This API must be evolved in compatible ways. If these are duplicates, then either one must be removed, or one must be defined in terms of the other, e.g. as an alias.